test/lifecycle_test.dart

```diff
@@ -145,11 +147,51 @@ main() {
         expect(
             component.lifecycleCalls,
             equals([
-              matchCall('componentWillUpdate', args: [expectedProps, initialState], state: initialState),
               matchCall('render', state: initialState),
-              matchCall('componentDidUpdate', args: [expectedProps, initialState], state: initialState),
+              matchCall('getSnapshotBeforeUpdate', args: [expectedProps, initialState], state: initialState),
+              matchCall('componentDidUpdate', args: [expectedProps, initialState, null], state: initialState),
             ].where((matcher) => matcher != null).toList()));
       });
+
+      test(
+          'getSnapshotBeforeUpdate returns the same value being passed into '
+          'componentDidUpdate', () {
+        LifecycleTestHelper component = getDartComponent(
+            render(components2.LifecycleTest({'getSnapshotBeforeUpdate': (_, __, ___) => 'snapshot'})));
+
+        component.lifecycleCalls.clear();
+        component.setState({});
+
+        expect(
+            component.lifecycleCalls,
+            containsAllInOrder([
+              matchCall('getSnapshotBeforeUpdate'),
+              matchCall('componentDidUpdate', args: [anything, anything, 'snapshot']),
+            ]));
+      });
+
+      test(
+          'does not jsify the return value of getSnapshotBeforeId, and '
+          'passes the same instance of that value to componentDidUpdate', () {
+        List<String> expectedSnapshot = [
+          'List Item 1',
+          'List Item 2',
+          'List Item 3',
+        ];
+
+        LifecycleTestHelper component = getDartComponent(
+            render(components2.LifecycleTest({'getSnapshotBeforeUpdate': (_, __, ___) => expectedSnapshot})));
+
+        component.lifecycleCalls.clear();
+        component.setState({});
+
+        expect(
+            component.lifecycleCalls,
+            containsAllInOrder([
+              matchCall('getSnapshotBeforeUpdate'),
+              matchCall('componentDidUpdate', args: [anything, anything, same(expectedSnapshot)]),
```

Comment:
I dont think the `same` does anything here... I know @greglittlefield-wf said https://github.com/cleandart/react-dart/pull/169/files#r266086539 but @greglittlefield-wf doesnt the `containsAllInOrder` check this?

Reply:
This test ensures that the same instance is passed back, which we need `same` for.